Gems/LyShine/Code/Tests/AnimationTest.cpp

```diff
@@ -81,6 +88,20 @@ namespace UnitTest
             m_timeSystem = AZStd::make_unique<UnitTest::AnimationTestStubTimer>();
         }
 
+        // override and only include system components required for tests.
+        AZ::ComponentTypeList GetRequiredSystemComponents() const override
+        {
+            return AZ::ComponentTypeList{
+                azrtti_typeid<AZ::MemoryComponent>(),
+                azrtti_typeid<AZ::AssetManagerComponent>(),
+                azrtti_typeid<AZ::JobManagerComponent>(),
+                azrtti_typeid<AZ::StreamerComponent>(),
+                azrtti_typeid<AZ::SliceSystemComponent>(),
+                azrtti_typeid<AzFramework::GameEntityContextComponent>(),
+                azrtti_typeid<AzFramework::AssetSystem::AssetSystemComponent>(),
+            };
+        }
+
```

Comment:
This solution would require keeping up with changes to the list of required components from the `AzFramework::Application`, which means if it changes, then this file needs to be updated.

A better solution is to override the `AZ::ComponentApplication::Create` function to add a call to DisableSaveOnFinalize to prevent the UserSettings Component from saving.

```suggestion
        // override and only include system components required for tests.
        AZ::Entity* Create(const AZ::ComponentApplication::Descriptor* descriptor
            const AZ::ComponentApplication::StartupParameters& startupParameters = {}) override
        {
            // Call base Application Create function
            AzFramework::Application::Create(descriptor, startupParameters);
            
            // Without this, the user settings component would attempt to save on finalize/shutdown. Since the file is
            // shared across the whole engine, if multiple tests are run in parallel, the saving could cause a crash
            // in the unit tests.
             AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
        }

```

Reply:
Made the requested changes so test doesn't need to be updated when new required system components become necessary for the test to run.